File: OmegaUp/6384. Búsqueda de Patrones en Cadenas.java
import java.util.Scanner;
public class Main {
  public static void main ( String[] args ) {
    Scanner scanner = new Scanner ( System.in );
    String phrase = scanner.next();
    String pattern = scanner.next();
    for ( int i = 0; i < phrase.length(); i++ ) {
      if ( phrase.charAt(i) == pattern.charAt(0) ) {
        if ( phrase.length() - i + 2 < pattern.length() )
          continue;
        boolean temp = true;
        for ( int j = 0; j < pattern.length(); j++ ) 
          if ( pattern.charAt(j) != phrase.charAt(i+j) ) {
            temp = false;
            break;
          }
        if ( temp ) 
          System.out.println ( i + 1 );
      }
    }
  }
}

File: OmegaUp/6384. Búsqueda de Patrones en Cadenas.cpp
#include <iostream>
using namespace std;
int main () {
  string text;
  cin >> text;
  string pattern;
  cin >> pattern;
  for ( int i = 0; i < text.length(); i++ ) {
    if ( text[i] == pattern[0] ) {
      if ( text.length() - i + 2 < pattern.length() )
        continue;
      bool temp = true;
      for ( int j = 0; j < pattern.length(); j++ ) 
        if ( pattern[j] != text[i+j] ) {
          temp = false;
          break;
        }
      if ( temp )
        cout << i + 1 << endl;
    }
  }
}
